pytorch_lightning/loggers/mlflow.py

```diff
@@ -183,6 +189,18 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
 
             self.experiment.log_metric(self.run_id, k, v, timestamp_ms, step)
 
+    @rank_zero_only
+    def log_figure(self, name: str, figure: plt.figure, step: Optional[int] = None, close: bool = True) -> None:
+
+        filename = self.save_dir + '/' + name + f"_step_{step}" + self._figure_file_extension
+        figure.savefig(filename)
+        self.experiment.log_artifact(self.run_id, filename, artifact_path="figure_" + name)
+
+        Path(filename).unlink()  # delete temporary file
```

Comment:
do we need to wait here before this was uploaded (upload may be async, not sure)?

Also if this is temporary, can we consider this to be saved in /tmp?

Reply:
That's a good point, couldn't find anything in the docs of mlflow ...
I don't like this temporary thing anyways but could not come up with a better solution

Reply:
Changed to temporary file